src/Tasks/Unzip.cs

```diff
@@ -0,0 +1,211 @@
+                }
+
+                try
+                {
+                    using (FileStream stream = new FileStream(sourceFile.ItemSpec, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 0x1000, useAsync: false))
+                    {
+                        using (ZipArchive zipArchive = new ZipArchive(stream, ZipArchiveMode.Read, leaveOpen: false))
+                        {
+                            try
+                            {
+                                Extract(sourceFile, zipArchive, destinationDirectory);
+                            }
+                            catch (Exception e)
+                            {
+                                // Unhandled exception in Extract() is a bug!
+                                Log.LogErrorFromException(e, showStackTrace: true);
+                                return false;
+                            }
+                        }
+                    }
+                }
+                catch (Exception e)
+                {
+                    // Should only be thrown if the archive could not be opened (Access denied, corrupt file, etc)
+                    Log.LogErrorFromResources("Unzip.ErrorCouldNotOpenFile", e.Message);
+                }
+            }
+
+            return !_canceling && !Log.HasLoggedErrors;
+        }
+
+        /// <summary>
+        /// Extracts all files to the specified directory.
+        /// </summary>
+        /// <param name="sourceTaskItem">The original <see cref="ITaskItem"/> containing details about the source file.</param>
+        /// <param name="sourceArchive">The <see cref="ZipArchive"/> containing the files to extract.</param>
+        /// <param name="destinationDirectory">The <see cref="DirectoryInfo"/> to extract files to.</param>
+        private void Extract(ITaskItem sourceTaskItem, ZipArchive sourceArchive, DirectoryInfo destinationDirectory)
+        {
+            foreach (ZipArchiveEntry zipArchiveEntry in sourceArchive.Entries.TakeWhile(i => !_canceling))
+            {
+                FileInfo destinationPath = new FileInfo(Path.Combine(destinationDirectory.FullName, zipArchiveEntry.FullName));
+
+                if (!destinationPath.FullName.StartsWith(destinationDirectory.FullName, StringComparison.OrdinalIgnoreCase))
+                {
+                    // ExtractToDirectory() throws an IOException for this but since we're extracting one file at a time
+                    // for logging and cancellation, we need to check for it ourselves.
+                    Log.LogErrorFromResources("Unzip.ErrorExtractingResultsInFilesOutsideDestination", destinationPath.FullName, destinationDirectory.FullName);
+                    continue;
+                }
+                
+                TaskItem taskItem = new TaskItem(EscapingUtilities.Escape(destinationPath.FullName));
+
+                sourceTaskItem.CopyMetadataTo(taskItem);
+
+                _destinationFiles.Add(taskItem);
```

Comment:
Do we always want this? Seems like it could be a giant list in some cases and use up a lot of memory and may or may not be needed? The output folder might be enough?

Reply:
Its what `<Copy />` does so I though I'd ... :expressionless: ... 😎 _copy_ it.

Technically you could copy a ton of files as well 

```xml
<ItemGroup>
  <Files Include="Folder\**" />
</ItemGroup>

<Copy
  SourceFiles="@(Files)"
  DestinationFolder="Foo"
  />
```

And we'll waste all of the memory.  What should I do?

Reply:
@jeffkl Would it save memory to make _unzippedFiles and _destinationFiles a collection of tuples of sourceTaskItem and destinationPath.FullName? That way if you access the output properties DestinationFiles or UnzippedFiles you could build the full list of TaskItems on demand?

Reply:
We decided to just remove it.  If necessary, we can add the functionality later.